src/scopes.js

```diff
@@ -56,3 +61,47 @@ exports.scopeMatch = function(scopePatterns, scopesets) {
     });
   });
 };
+
+/**
+ * Finds scope intersections between scopePatterns and scope-set
+ *
+ * Note that scope-set is an array of arrays of strings on negation-free
+ * disjunctive normal form.
+ *
+ */
+exports.scopeIntersection = (scopePatterns, scopesets) => {
+  exports.validateScopeSets(scopesets);
+  validateScopePatterns(scopePatterns);
+
+  const junctions = [];
+  const scopes = Array.prototype
+    .concat(...scopesets)
+    .filter(scope => scope);
```

Comment:
it looks like this is flattening scopesets?  So I think you could just use
```
const scopes1 = scopeset1.filter(s => s);
const scopes2 = scopeset2.filter(s => s);
```
then use the same loop you have below.